js/runtime-simple.js

```diff
@@ -199,5 +206,78 @@
             }
         },
 
+        /*
+         * The underlying JavaScript object is the same object that is passed
+         * to the getCurrentLocation callback.
+         */
+        geolocation: {
+            /* Synchronous "get current location" */
+            currentLocation: function () {
+                return util.geolocation.currentLocation;
+            },
+            /* Asynchronous update event. Context. */
+            whenLocationUpdated: function(args, containers) {
+                var currentScope = this,
+                    steps = containers[0];
+
+                Event.on(window, 'locationchanged', null, function (event) {
+                    // TODO: probably factor out augmenting scope and running
+                    // the block stuff to somewhere else.
+                    steps.forEach(function (block) {
+                        block.run(currentScope);
+                    });
+                });
+            },
+            // Returns the distance between two points in meters.
+            // taken from http://www.movable-type.co.uk/scripts/latlong.html
+            // Using the haversine formula.
+            distanceBetween: function (p1, p2) {
+                var R = 6371000; // m
+                var lat1 = p1.coords.latitude;
+                var lon1 = p1.coords.longitude;
+                var lat2 = p2.coords.latitude;
+                var lon2 = p2.coords.longitude;
+
+                var φ1 = lat1.toRadians();
+                var φ2 = lat2.toRadians();
+                var Δφ = (lat2-lat1).toRadians();
+                var Δλ = (lon2-lon1).toRadians();
+
+                var a = Math.sin(Δφ/2) * Math.sin(Δφ/2) +
+                        Math.cos(φ1) * Math.cos(φ2) *
+                        Math.sin(Δλ/2) * Math.sin(Δλ/2);
+                var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a));
+
+                return R * c;
+            },
+            /* Returns latitude in degrees. */
+            // TODO: should this return a "degrees" object?
+            latitude: function (location) {
+                return location.coords.latitude;
+            },
+            /* Returns longitude in degrees. */
+            // TODO: should this return a "degrees" object?
+            longitude: function (location) {
+                return location.coords.longitude;
+            },
+            /* Returns altitude as a unit? */
+            altitude: function (location) {
+                return location.coords.altitude;
```

Comment:
What do we do when altitude is `null`? According to [the spec](http://dev.w3.org/geo/api/spec-source.html#coordinates_interface), if the device can't come up with it, it's gotta be null. But how do we present that to Waterbear users?